python/tests/test_gaussianbeam.py

```diff
@@ -76,10 +76,11 @@ def gaussian_beam(self, rot_angle):
             f"ratio of the Gaussian beam energy at the focus over the maximum beam energy for the entire cell: {frac}"
         )
 
-        self.assertGreater(frac, 0.99)
+        self.assertGreater(frac, 0.98)
 
     def test_gaussian_beam(self):
-        self.gaussian_beam(-40)
+        self.gaussian_beam(-40, mp.GaussianBeam2DSource)
+        self.gaussian_beam(-40, mp.GaussianBeam3DSource)
```

Comment:
Since this will just trigger the same 2d test as the previously line, to avoid the redundant calculation it might be better to just test that this function call correctly raises a warning. The procedure for how to do this is described in https://docs.python.org/3/library/warnings.html#testing-warnings.

Reply:
Nevermind. This is correct since this line will execute the 2d simulation using the cross section of the 3d Gaussian beam (the existing test).

Reply:
We can still test the deprecation warning if we wish. It could look like 

```python
    def test_gaussian_beam(self):

        # 2D Gaussian beam test
        self.gaussian_beam(-40, mp.GaussianBeam2DSource)
        self.gaussian_beam(-40, mp.GaussianBeam3DSource)

        # 3D Gaussian beam deprecation warning test
        with warnings.catch_warnings(record=True) as w:
            warnings.simplefilter("always")
            self.gaussian_beam(-40, mp.GaussianBeamSource)
            self.assertTrue(len(w) == 1)
            self.assertTrue(issubclass(w[-1].category, DeprecationWarning))
            self.assertTrue("deprecated" in str(w[-1].message))
```

Should I include the addition or keep it out? It would add one extra meep run to the test cases. Alternatively we could exclude `self.gaussian_beam(-40, mp.GaussianBeam3DSource)` since calling GaussianBeamSource will test that case.